### DISCUSSION TITLE
Reactivity extended to components defined in src imported into markdown files?

### DISCUSSION THREAD

**Discussion by DevinBayly** [observablehq/framework#1711] [Q&A] [answered]
Hello there,

I'm attempting to follow some of the project structure best practices that are laid out in https://observablehq.com/framework/project-structure, and I'd love to define a chart element that has interactive features to it within the `src` folder and import it into a markdown file. The plan is that when someone interacts with the chart, then a variable updates that other parts of the markdown file can react to. Unfortunately as I have it written now I don't see any changes that come after interaction. I also explore the [`Mutable` ](https://observablehq.com/framework/reactivity#mutables) option and the same thing happened. 

Can someone advise how to extend reactivity into components defined within `src` files that are imported into a markdown page?

**Comment by DevinBayly** [comment]
Oh on second thought it sounds like [Generators ](https://observablehq.com/framework/reactivity#generators)might be more like the option I'm looking for. I'll check into this, but if anyone has a more simple/framework idiomatic solution I'm all ears.

**Comment by mbostock** [comment, marked as answer]
Sounds like you want to use `view` (or `Generators.input`). If your chart emits *input* events and exposes a `chart.value`, you can use with it `view` to declare a reactive value that is driven by interacting with a chart. There’s an example of this in the docs.

**Reply by DevinBayly** [reply]
that sounds like exactly what I'm looking for. The stuff I was getting setup with `observe` and `notify` seemed like it wasn't quite right. 

Are you talking about the section of the [reactive page on inputs](https://observablehq.com/framework/reactivity#inputs) where the `view` is used on a `Plot.dot` chart?

I think I recall using the input event emitting on some older hq documents so that will probably be enough for me to get going with this approach.

**Reply by mbostock** [reply]
Yep, that’s the example I was thinking of.
